### PR TITLE
Pull in upstream updates

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description

This pull request pulls in upstream changes from [cisagov/skeleton-ansible-role](https://github.com/cisagov/skeleton-ansible-role).

## 💭 Motivation and Context

This Ansible role needs to be brought into the light so it can reap the benefits of [cisagov/action-lineage](https://github.com/cisagov/action-lineage) and all the other goodies we have added in the past several months.

Modernize or be left behind!

## 🧪 Testing

All pre-commit hooks and molecule tests pass.

## 🚥 Types of Changes

- Upstream updates

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** document.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by jsf9k** [comment]
The builds are failing for interesting reasons, at least on my local machine.  My local system runs Linux 5.9.1, and that activates [a bug with certain versions of `systemd`](https://github.com/systemd/systemd/pull/16424).  Debian 10 and Ubuntu Xenial and Focal all currently run versions of `systemd` that exhibit this bug.  Our other test OSs run versions of `systemd` that are either too old or too new to contain the bug.

[A PR for `ansible` has been merged that works around the bug](https://github.com/ansible/ansible/pull/72337), but [the PR to backport those changes to Ansible 2.10](https://github.com/ansible/ansible/pull/72347) has not yet been merged.

Therefore we are stuck with broken builds for Debian 10 and Ubuntu Xenial and Focal on some local systems (including mine) until one of the following happens:
* These OSs offer a version of `systemd` (version 245.7 or higher) that does not have the bug
* ansible/ansible#72347 is merged and a new version of Ansible 2.10 is released to PyPI

**Comment by jsf9k** [comment]
After hours of debugging, I have determined that the build is currently failing in GitHub Actions for all Debian-based systems because:
* apparmor is enabled in the host kernel
* The clam/freshclam installation installs an apparmor profile for freshclam
* The apparmor profile is somehow disallowing freshclam from using DNS

I think the solution here is to check if:

* apparmor is enabled (`cat /sys/module/apparmor/parameters/enabled` will yield "Y" if and only if apparmor is enabled)
* `/etc/apparmor.d/usr.bin.freshclam` exists

If both are true then we should disable that profile:

```console
$ ln --symbolic /etc/apparmor.d/usr.bin.freshclam /etc/apparmor.d/disable/
$ apparmor_parser --remove /etc/apparmor.d/disable/usr.bin.freshclam
```

or

```console
$ aa-disable /usr/bin/freshclam
```

**Comment by jsf9k** [comment]
> If both are true then we should disable that profile:

As explained in 5aeac40, it is better to use `aa-complain` here instead of `aa-disable`.  In complain mode the AppArmor policy violations are allowed but logged, so there is still a record of them.  Furthermore, you can run `aa-complain` multiple times without an error, which means we no longer need to ignore errors from the `aa-*` step in the playbook.

File: molecule/default/prepare.yml [hunk 1]
```diff
@@ -1,29 +1,71 @@
...
+#
+# Debian and Kali ship with an AppArmor profile for /usr/bin/freshclam
+# that, when applied to /usr/bin/freshclam running in a Docker
+# container, does not allow /usr/bin/freshclam to query DNS.  (The
+# same profile works as expected when run outside of Docker on a
+# Debian or Kali AMI.)  This obviously interferes with the downloading
+# of updated ClamAV databases and breaks this Ansible role.  Until we
+# can figure out exactly what is happening, the only remedy I have
+# found is to switch the AppArmor profile for /usr/bin/freshclam to
+# complain mode (or disable it entirely).
```
Comment (dav3r) [inline molecule/default/prepare.yml:21]: We should probably create an issue for this and link to it in this comment for posterity.

File: molecule/default/prepare.yml [hunk 2]
```diff
@@ -1,29 +1,71 @@
...
+          # we want to ignore
+          ignore_errors: true
+          register: apparmor_enabled
+          tags:
+            - molecule-idempotence-notest
+        - name: |
+            Switch the usr.bin.freshclam AppArmor profile to complain \
+            mode
+          block:
+            # We do this so because we need the usr.bin.freshclam
```
Comment (dav3r) [inline molecule/default/prepare.yml]:
```suggestion
            # We do this because we need the usr.bin.freshclam
```

**Comment by hillaryj** [comment]
> which means we no longer need to ignore errors from the aa-* step in the playbook.

Which runbook is this, and has the step been updated (or will it be once this is merged)?

File: molecule/debian/molecule.yml [hunk 3]
```diff
@@ -60,6 +46,35 @@ provisioner:
...
+        #
+        # In auto_legacy mode Ansible will prefer the /usr/bin/python
+        # interpreter if it exists, while in auto mode it will prefer
+        # the Python interpreter specified in an internal map.  In
+        # most cases we only have Python3 installed, and the internal
+        # map specifies /usr/bin/python3, so it is immaterial whether
+        # we use auto_legacy or auto.
+        #
+        # The one place where it _does_ matter is our Debian9 AKA CyHy
+        # instances.  Here the /usr/bin/python _and_ /usr/bin/python3
```
Comment (hillaryj) [inline molecule/debian/molecule.yml]:
```suggestion
        # instances, where the /usr/bin/python _and_ /usr/bin/python3
```

File: molecule/default/molecule.yml [hunk 4]
```diff
@@ -78,6 +75,35 @@ provisioner:
...
+        #
+        # In auto_legacy mode Ansible will prefer the /usr/bin/python
+        # interpreter if it exists, while in auto mode it will prefer
+        # the Python interpreter specified in an internal map.  In
+        # most cases we only have Python3 installed, and the internal
+        # map specifies /usr/bin/python3, so it is immaterial whether
+        # we use auto_legacy or auto.
+        #
+        # The one place where it _does_ matter is our Debian9 AKA CyHy
+        # instances.  Here the /usr/bin/python _and_ /usr/bin/python3
```
Comment (hillaryj) [inline molecule/default/molecule.yml]:
```suggestion
        # instances, where the /usr/bin/python _and_ /usr/bin/python3
```

File: molecule/fedora/molecule.yml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        #
+        # In auto_legacy mode Ansible will prefer the /usr/bin/python
+        # interpreter if it exists, while in auto mode it will prefer
+        # the Python interpreter specified in an internal map.  In
+        # most cases we only have Python3 installed, and the internal
+        # map specifies /usr/bin/python3, so it is immaterial whether
+        # we use auto_legacy or auto.
+        #
+        # The one place where it _does_ matter is our Debian9 AKA CyHy
+        # instances.  Here the /usr/bin/python _and_ /usr/bin/python3
```
Comment (hillaryj) [inline molecule/fedora/molecule.yml]:
```suggestion
        # instances, where the /usr/bin/python _and_ /usr/bin/python3
```

File: molecule/ubuntu/molecule.yml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        #
+        # In auto_legacy mode Ansible will prefer the /usr/bin/python
+        # interpreter if it exists, while in auto mode it will prefer
+        # the Python interpreter specified in an internal map.  In
+        # most cases we only have Python3 installed, and the internal
+        # map specifies /usr/bin/python3, so it is immaterial whether
+        # we use auto_legacy or auto.
+        #
+        # The one place where it _does_ matter is our Debian9 AKA CyHy
+        # instances.  Here the /usr/bin/python _and_ /usr/bin/python3
```
Comment (hillaryj) [inline molecule/ubuntu/molecule.yml]:
```suggestion
        # instances, where the /usr/bin/python _and_ /usr/bin/python3
```

**Review by dav3r — APPROVED** [review]
Wow, thank you for banging your head against this thang and emerging victorious!  This all looks excellent to me.  Strongest of work! 💪 💪 💼 🏆

**Review by hillaryj — APPROVED** [review]
I added some comment-wording changes (one change, several times) as suggested changes. Other than that, excellent work! 🌮

**Comment by jsf9k** [comment]
> > which means we no longer need to ignore errors from the aa-* step in the playbook.
> 
> Which runbook is this, and has the step been updated (or will it be once this is merged)?

"Playbook", not "runbook".  I'm talking about [this step in the `prepare.yml` Ansible playbook](https://github.com/cisagov/ansible-role-clamav/blob/improvement/upstream-updates/molecule/default/prepare.yml#L69-L70).  As you can see, the step has already been updated.

**Comment by hillaryj** [comment]
Aha, perfect then!

**Review by hillaryj — APPROVED** [review]
✍🏻 ✅

File: molecule/default/prepare.yml (hunk 1 above)
Comment (jsf9k) [reply molecule/default/prepare.yml:21]: See #16.  The source code links in the body of the issue should be correct once this PR is merged.

**Comment by jsf9k** [comment]
> Wow, thank you for banging your head against this thang and emerging victorious! This all looks excellent to me. Strongest of work!

This is Sparta!!!

**Comment by jsf9k** [comment]
> Therefore we are stuck with broken builds for Debian 10 and Ubuntu Xenial and Focal on some local systems (including mine) until one of the following happens:
> 
> * These OSs offer a version of `systemd` (version 245.7 or higher) that does not have the bug
> * [ansible/ansible#72347](https://github.com/ansible/ansible/pull/72347) is merged and a new version of Ansible 2.10 is released to PyPI

[ansible/ansible#72347](https://github.com/ansible/ansible/pull/72347) was merged yesterday, but I don't _think_ it has yet worked its way into an Ansible release.  Soon!

**Comment by jsf9k** [comment]
> > Therefore we are stuck with broken builds for Debian 10 and Ubuntu Xenial and Focal on some local systems (including mine) until one of the following happens:
> > 
> > * These OSs offer a version of `systemd` (version 245.7 or higher) that does not have the bug
> > * [ansible/ansible#72347](https://github.com/ansible/ansible/pull/72347) is merged and a new version of Ansible 2.10 is released to PyPI
> 
> [ansible/ansible#72347](https://github.com/ansible/ansible/pull/72347) was merged yesterday, but I don't _think_ it has yet worked its way into an Ansible release. Soon!

Looks like ansible/ansible#72347 is in Ansible 2.10.4, which is out now.  I have verified that the systemd issue is now fixed on my local system.

**Pull request merged**